src/main/kotlin/graphics/scenery/controls/behaviours/MouseDrag.kt

```diff
@@ -0,0 +1,67 @@
+package graphics.scenery.controls.behaviours
+
+import graphics.scenery.BoundingGrid
+import graphics.scenery.Camera
+import graphics.scenery.Node
+import graphics.scenery.utils.LazyLogger
+import graphics.scenery.utils.extensions.plus
+import graphics.scenery.utils.extensions.times
+import org.scijava.ui.behaviour.DragBehaviour
+import kotlin.reflect.KProperty
+
+/**
+ * Drag nodes roughly along the view plane axis by mouse.
+ * Implements algorithm from https://forum.unity.com/threads/implement-a-drag-and-drop-script-with-c.130515/
+ *
+ * @author Jan Tiemann <j.tiemann@hzdr.de>
+ */
+open class MouseDrag(
+    protected val name: String,
+    protected val camera: () -> Camera?,
+    protected var debugRaycast: Boolean = false,
+    var ignoredObjects: List<Class<*>> = listOf<Class<*>>(BoundingGrid::class.java)
```

Comment:
This should be a val and a MutableList so elements could be added or removed at runtime, without reinitialising the property.

Reply:
Will there be so many changes that it is worth losing the thread-safety over it?